jax/interpreters/masking.py

```diff
@@ -253,33 +262,53 @@ def mul(coeff, mon):
   except:
     return coeff * mon
   else:
-    return  0 if coeff == 0 else mon if coeff == 1 else coeff * mon
+    return 0 if coeff == 0 else mon if coeff == 1 else coeff * mon
 
 
 abstract_arrays._DIMENSION_TYPES.add(Poly)
 
-
 class Mon(dict):
+  """Represents a multivariante monomial, such as n^3 * m."""
   def __hash__(self):
     return hash(frozenset(self.items()))
 
   def __str__(self):
-    return ' '.join('{}**{}'.format(k, v) if v != 1 else str(k)
-                    for k, v in sorted(self.items()))
+    return ' '.join(f'{key}^{exponent}' if exponent != 1 else str(key)
+                    for key, exponent in sorted(self.items()))
 
-  def __lt__(self, other):
-    # sort by total degree, then lexicographically on indets
+  def __lt__(self, other: 'Mon'):
```

Comment:
I find this strange, and hard to understand. IIUC, this is not the numeric comparison, just a sorting order for monomials based on their representation. 

The "-self.degree" along with ">" used in an implementation of "__lt__" is confusing.  

Is this needed only for _leading term? If so, wouldn't it be clearer if in that function we define this as sorting key without overloading __lt__?

Reply:
@gnecula I think what you suggest is not equivalent, see https://en.wikipedia.org/wiki/Monomial_order#Graded_reverse_lexicographic_order

I replaced the direction here since `x^2<x` (how it was before) is unintuitive.
Semantically, various other orders could be used without breaking poly division. I chose this order to allow creating string representations following `PolyTest.test_parse_spec` just by sorting.